homeassistant/components/miele/sensor.py

```diff
@@ -770,23 +770,16 @@ class MielePhaseSensor(MieleSensor):
     @property
     def native_value(self) -> StateType:
         """Return the state of the sensor."""
-        ret_val = STATE_PROGRAM_PHASE.get(self.device.device_type, {}).get(
-            self.device.state_program_phase
+        return (
+            PROGRAM_PHASE[self.device.device_type](self.device.state_program_phase).name
+            if self.device.device_type in PROGRAM_PHASE
+            else None
         )
-        if ret_val is None:
-            _LOGGER.debug(
-                "Unknown program phase: %s on device type: %s",
-                self.device.state_program_phase,
-                self.device.device_type,
-            )
-        return ret_val
 
     @property
     def options(self) -> list[str]:
         """Return the options list for the actual device type."""
-        return sorted(
-            set(STATE_PROGRAM_PHASE.get(self.device.device_type, {}).values())
-        )
+        return sorted(PROGRAM_PHASE[self.device.device_type].keys())
```

Comment:
How come we no longer need to guard with `dict.get`?

Reply:
We will never get this far if there is an unsupported device_type, i.e. an entirely new type of appliance. I think it is better to get a full exception in the log than silently setting it up with an empty option list.